### PR TITLE
Fix SCIM query params

### PR DESCRIPTION
**Pull request by exageraldo**
Fixes: #2679

### CODE REVIEW

File: github/github_test.go
Comment (gmlewis) [inline github/github_test.go:175]:
```suggestion
	got, err := addOptions(url, v)
```

File: github/github_test.go
Comment (gmlewis) [inline github/github_test.go:181]:
```suggestion
		t.Errorf("addOptions(%q, %#v) returned %v, want %v", url, v, got, want)
```

File: github/scim_test.go
Comment (gmlewis) [inline github/scim_test.go:433]: Can we please keep this line:

```go
testJSONMarshal(t, &ListSCIMProvisionedIdentitiesOptions{}, `{}`)
```

for consistency?  We always test to see how an empty struct gets mashaled.

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/scim_test.go
Comment (exageraldo) [reply github/scim_test.go:435]: Done! ✅

Comment (exageraldo) [reply github/scim_test.go:435]: Since the `ListSCIMProvisionedIdentitiesOptions` struct has no json tag (url only), the string/json fields should be the same as the structure fields, and no field should be omitted if empty.

I think the expected value should be something like this:

```go
testJSONMarshal(t, &ListSCIMProvisionedIdentitiesOptions{}, `{
	"StartIndex": null,
	"Count": null,
	"Filter": null
}`)
```
I just created a new issue so we can talk more about this subject. #2699

File: github/github_test.go
Comment (exageraldo) [inline github/github_test.go:172]: Since the structures passed in must have "url" tags in the fields (more in [addOptions](https://github.com/google/go-github/blob/master/github/github.go#L286)), is it valid to test if the fields have this tag?

Something like this:

```go
st := reflect.TypeOf(v)
for i := 0; i < st.NumField(); i++ {
	field := st.Field(i)
	if alias, ok := field.Tag.Lookup("url"); ok {
		if alias == "" {
			// t.Errorf("...")
		}
	} else {
		// t.Errorf("...")
	}
}
```

Comment (gmlewis) [reply github/github_test.go:172]: Sure.  Please add a comment before line 172 that describes this requirement so that people will not start adding `testAddOptions` to every test.

In fact, let's make the function name more descriptive.  How about `testAddURLOptions` ?

Comment (gmlewis) [reply github/github_test.go:172]: Or maybe `testAdURLTags` ?

File: github/scim_test.go
Comment (gmlewis) [inline github/scim_test.go:433]: OK, now I understand why you removed this before, as it really is no longer relevant.
Feel free to delete this if you wish, either in this PR or in your upcoming one.

**Review by gmlewis — APPROVED** [review]
Thank you, @exageraldo !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
